backend/app/app.go

```diff
@@ -248,117 +94,118 @@ func (app *App) registerHandlers() {
-				deploymentGroup.DELETE("", app.handlers.HandleDeleteAllDeployments)
-				deploymentGroup.GET("/:name", app.handlers.HandleGetDeployment)
-				deploymentGroup.GET("/:name/kubeconfig", app.handlers.HandleGetKubeconfig)
-				deploymentGroup.DELETE("/:name", app.handlers.HandleDeleteCluster)
-				deploymentGroup.POST("/:name/nodes", app.handlers.HandleAddNode)
-				deploymentGroup.DELETE("/:name/nodes/:node_name", app.handlers.HandleRemoveNode)
+				deploymentGroup.POST("", app.deploymentHandler.HandleDeployCluster)
+				deploymentGroup.GET("", app.deploymentHandler.HandleListDeployments)
+				deploymentGroup.DELETE("", app.deploymentHandler.HandleDeleteAllDeployments)
+				deploymentGroup.GET("/:name", app.deploymentHandler.HandleGetDeployment)
+				deploymentGroup.GET("/:name/kubeconfig", app.deploymentHandler.HandleGetKubeconfig)
+				deploymentGroup.DELETE("/:name", app.deploymentHandler.HandleDeleteCluster)
+				deploymentGroup.POST("/:name/nodes", app.deploymentHandler.HandleAddNode)
+				deploymentGroup.DELETE("/:name/nodes/:node_name", app.deploymentHandler.HandleRemoveNode)
 			}
 
 			notificationGroup := deployerGroup.Group("/notifications")
 			{
-				notificationGroup.GET("", app.handlers.GetAllNotificationsHandler)
-				notificationGroup.GET("/unread", app.handlers.GetUnreadNotificationsHandler)
-				notificationGroup.PATCH("/read-all", app.handlers.MarkAllNotificationsReadHandler)
-				notificationGroup.DELETE("", app.handlers.DeleteAllNotificationsHandler)
-				notificationGroup.PATCH("/:notification_id/read", app.handlers.MarkNotificationReadHandler)
-				notificationGroup.PATCH("/:notification_id/unread", app.handlers.MarkNotificationUnreadHandler)
-				notificationGroup.DELETE("/:notification_id", app.handlers.DeleteNotificationHandler)
+				notificationGroup.GET("", app.notificationHandler.GetAllNotificationsHandler)
+				notificationGroup.GET("/unread", app.notificationHandler.GetUnreadNotificationsHandler)
+				notificationGroup.PATCH("/read-all", app.notificationHandler.MarkAllNotificationsReadHandler)
+				notificationGroup.DELETE("", app.notificationHandler.DeleteAllNotificationsHandler)
+				notificationGroup.PATCH("/:notification_id/read", app.notificationHandler.MarkNotificationReadHandler)
+				notificationGroup.PATCH("/:notification_id/unread", app.notificationHandler.MarkNotificationUnreadHandler)
+				notificationGroup.DELETE("/:notification_id", app.notificationHandler.DeleteNotificationHandler)
 			}
 		}
 	}
 	app.router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler))
 }
 
 func (app *App) StartBackgroundWorkers() {
-	go app.handlers.MonthlyInvoicesHandler()
-	go app.handlers.TrackUserDebt(app.gridClient)
-	go app.handlers.MonitorSystemBalanceAndHandleSettlement()
-	go app.handlers.TrackClusterHealth()
-	go app.handlers.TrackReservedNodeHealth(app.notificationService, app.handlers.proxyClient)
+	go app.invoiceHandler.MonthlyInvoicesHandler()
```

Comment:
should we pass context to all of these go routines for cancellation?

we can address in a separate issue if you feel it's not in the scope of the PR